packages/neon_framework/lib/src/utils/findable.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:collection/collection.dart';
+
+/// Interface to be implemented by classes that want to use the [FindExtension.find] and [FindExtension.tryFind] methods.
+abstract interface class Findable {
+  /// Unique ID of the element.
+  String get id;
+}
+
+extension FindExtension<T extends Findable> on Iterable<T> {
+  /// Returns the first element matching [id] by [Findable.id].
+  ///
+  /// If no element was found `null` is returned.
+  T? tryFind(final String? id) => firstWhereOrNull((final e) => e.id == id);
```

Comment:
I'd rather not support nullable parameters but we already use them throughout the code so let's early abort.

```suggestion
  T? tryFind(final String? id) {
	if(id == null){
		return null;
	}
	
        return firstWhereOrNull((final e) => e.id == id);
  }
```